src/DataPersister/MessengerDataPersister.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\DataPersister;
+
+use ApiPlatform\Core\Metadata\Resource\Factory\ResourceMetadataFactoryInterface;
+use ApiPlatform\Core\Util\ClassInfoTrait;
+use Symfony\Component\Messenger\Envelope;
+use Symfony\Component\Messenger\MessageBusInterface;
+
+/**
+ * Dispatches the given resource using the message bus of Symfony Messenger.
+ *
+ * @experimental
+ *
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ */
+final class MessengerDataPersister implements DataPersisterInterface
+{
+    use ClassInfoTrait;
+
+    private $resourceMetadataFactory;
+    private $messageBus;
+
+    public function __construct(ResourceMetadataFactoryInterface $resourceMetadataFactory, MessageBusInterface $messageBus)
+    {
+        $this->resourceMetadataFactory = $resourceMetadataFactory;
+        $this->messageBus = $messageBus;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function supports($data): bool
+    {
+        return true === $this->resourceMetadataFactory->create($this->getObjectClass($data))->getAttribute('messenger');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function persist($data)
+    {
+        $this->messageBus->dispatch($data);
```

Comment:
not a return here?

Reply:
It needs to return the data: https://github.com/api-platform/core/blob/master/src/DataPersister/ChainDataPersister.php#L55 dispatch returns an envelope IIRC